bcipy/gui/gui_main.py

```diff
@@ -196,6 +203,58 @@ def widgets(self) -> List[QWidget]:
         return [self.label_widget, self.help_tip_widget, self.control, self]
 
 
+class IntegerInput(FormInput):
+    """FormInput to select a integer value using a spinbox for selection. Help text is not
+    displayed for spinbox items.
+
+    Parameters:
+    ----------
+        label - form label.
+        value - initial value.
+    """
+
+    def __init__(self, **kwargs):
+        super(IntegerInput, self).__init__(**kwargs)
+
+    def init_control(self, value):
+        """Override FormInput to create a spinbox."""
+        spin_box = QSpinBox()
+        spin_box.setMaximum(100000)
+        return spin_box
+
+    def cast_value(self) -> str:
+        """Override FormInput to return an integer value."""
+        if self.control:
+            return int(self.control.text())
+        return None
+
+
+class FloatInput(FormInput):
+    """FormInput to select a float value using a spinbox for selection. Help text is not
+    displayed for spinbox items.
+
+    Parameters:
+    ----------
+        label - form label.
+        value - initial value.
+    """
+
+    def __init__(self, **kwargs):
+        super(FloatInput, self).__init__(**kwargs)
+
+    def init_control(self, value):
+        """Override FormInput to create a spinbox."""
+        spin_box = QDoubleSpinBox()
+        spin_box.setMaximum(100000)
+        return spin_box
+
+    def cast_value(self) -> float:
+        """Override FormInput to return as a float."""
+        if self.control:
+            return float(self.control.text())
+        return None
+
+
```

Comment:
It might be worth adding a `cast_value` to the BoolInput as well.

Reply:
I do think we should add it in the future, for this ticket, it would add another item to address for writing to json. If I cast to a boolean and dump it, it would store as:

`{"key": true}` instead of `{"key": "true"}`